app/Http/Controllers/Service/Admin/VouchersController.php

```diff
@@ -45,69 +41,109 @@ public function create()
     /**
      * Store Voucher range.
      *
-     * @param  \Illuminate\Http\Request  $request
-     * @return \Illuminate\Http\Response
+     * @param AdminNewVoucherRequest $request
+     * @return RedirectResponse
      */
-    public function storeBatch(Request $request)
+    public function storeBatch(AdminNewVoucherRequest $request)
     {
-        $new_vouchers = [];
-
-        $sponsor_ids = Sponsor::all()->pluck('id')->toArray();
-
-        $voucher_rules = [
-            'sponsor_id' => [
-                'required',
-                Rule::in($sponsor_ids)
-            ],
-            'start' => 'required|integer|between:1,99999999',
-            'end' => [
-                'required',
-                'integer',
-                'between:1,99999999',
-                'ge_field:start'
-            ],
-        ];
-
-        $messages = [
-            'in' => 'The :attribute must be a valid selection',
-            'ge_field' => 'The :attribute must be greater than or equal to :field'
-        ];
-
-        Validator::make($request->all(), $voucher_rules, $messages)->validate();
-
-        $codes = range($request['start'], $request['end']);
-
-        $sponsor = Sponsor::find($request['sponsor_id'])->first();
-        $shortcode = $sponsor->shortcode;
+        // Setup some variables
+        $input = $request->all();
+        $user_id = auth()->id();
```

Comment:
Do you want an abort if no user?

Reply:
in this case auth()->id being null isn't a showstopper, also, should always be auth'd by this point